examples/gridfs-upload.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/**
+ * This example demonstrates how to upload and download files from a stream with GridFS.
+ */
+
+declare(strict_types=1);
+
+namespace MongoDB\Examples;
+
+use MongoDB\BSON\ObjectId;
+use MongoDB\Client;
+
+use function assert;
+use function fclose;
+use function fopen;
+use function fwrite;
+use function getenv;
+use function rewind;
+
+use const PHP_EOL;
+use const STDOUT;
+
+require __DIR__ . '/../vendor/autoload.php';
+
+$client = new Client(getenv('MONGODB_URI') ?: 'mongodb://127.0.0.1/');
+
+$gridfs = $client->selectDatabase('test')->selectGridFSBucket();
+
+// Create an in-memory stream, this can be any stream source like STDIN or php://input for web requests
+$stream = fopen('php://temp', 'w+');
+fwrite($stream, 'Hello world!');
+rewind($stream);
```

Comment:
Is `rewind()` necessary? `Bucket::uploadFromStream()` uses [`stream_copy_to_stream()`](https://www.php.net/stream_copy_to_stream), the docs for which are actually a bit ambiguous because the `$offset` parameter defaults to `0`.

> Makes a copy of up to length bytes of data from the current position (or from the offset position, if specified) in from to to. If length is null, all remaining content in from will be copied.

Looking at the [implementation](https://github.com/php/php-src/blob/php-8.3.1/ext/standard/streamsfuncs.c#L517), it appears that `$offset=0` actually has no effect. The offset is only used to seek the `$from` stream when the value is > 0. That suggests that we can't even explicitly pass `0` to copy from the beginning of `$from`.

To that end, I suppose this `rewind()` _is_ necessary.

Reply:
Streams are generally consumed from the current cursor location. I have to rewind to get the file contents from the beginning.